Sources/PackageGraph/DependencyResolver.swift

```diff
@@ -171,6 +230,9 @@ public struct PackageContainerConstraint<T: PackageContainerIdentifier>: CustomS
         /// The requirement is specified by the version set.
         case versionSet(VersionSetSpecifier)
 
+        /// The requirement is specified by the revision.
+        case revision(String)
```

Comment:
This would be the place to document exactly what this means -- that it is an arbitrary string which must match and be totally equal to be valid. I think there is a probably a good generalization of this concept, but I don't see a need to tackle it now. That said, let's completely document the behavior.